lib/OpenQA/Utils.pm

```diff
@@ -224,6 +224,27 @@ sub needle_info {
     $needle->{name}      = $name;
     $needle->{distri}    = $distri;
     $needle->{version}   = $version;
+
+    # After change the needle's properties form to hash, support backward compatibility
+    return $needle unless $needle->{properties};
+
+    my @properties;
```

Comment:
Looks like this for-loop is meant to turn string-workaround-properties into HASH-workaround-properties. Maybe a comment at the top should explain that.

Reply:
I will add comments at the top.